cli/src/main/java/denominator/cli/Denominator.java

```diff
@@ -426,6 +428,32 @@ void overrideFromEnv(Map<String, String> env) {
       }
     }
 
+    static void setProxyFromEnv() {
+      setProtocolProxyFromEnv("http");
+      setProtocolProxyFromEnv("https");
+    }
+
+    static void setProtocolProxyFromEnv(String proto) {
+      String lowerProto = Ascii.toLowerCase(proto);
+      String envProxy = System.getenv(Ascii.toUpperCase(proto) + "_PROXY");
```

Comment:
ex.
Advice on test.


the happy path test asserts that `System.getProperty` returns valid host and port, and host and default port

the failure test asserts that System.exit was called and that System.err message logged.

http://stefanbirkner.github.io/system-rules/
```java

// use this to reset the system rules.
@Rule
public final RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();

// Use this to make sure the message logged
@Rule
public final SystemErrRule systemErrRule = new SystemErrRule().enableLog();

// use this to test exit occurred on malformed
@Rule
public final ExpectedSystemExit exit = ExpectedSystemExit.none();
```